docs/src/account/components.md

```diff
@@ -9,180 +9,201 @@ Account components are reusable units of functionality that define a part of an
 
 As an example, consider a typical wallet account, capable of holding a user's assets and requiring authentication whenever assets are added or removed. Such an account can be created by merging a `BasicWallet` component with an `RpoFalcon512` authentication component. The basic wallet does not need any storage, but contains the code to move assets in and out of the account vault. The authentication component holds a user's public key in storage and additionally contains the code to verify a signature against that public key. Together, these components form a fully functional wallet account.
 
-## Account Component templates
+## Account Component schemas
 
-An account component template is a description of an account component and contains all the
-information needed to initialize it.
+An account component schema describes a reusable piece of account functionality and captures
+everything required to initialize it. The schema encapsulates the component's **metadata**, its
+code, and how its storage should be laid out and typed.
 
-Specifically, a template specifies a component's **metadata** and its **code**.
-
-Once defined, a component template can be instantiated to an account component, which can then be
+Once defined, a component schema can be instantiated to an account component, which can then be
 merged to form the account's `Code` and `Storage`.
 
 ## Component code
 
-The template's code defines a library of functions that can read and write to the storage slots of the component.
+The component's code defines a library of functions that can read and write to the storage slots specified by the schema.
 
 ## Component metadata
 
 The component metadata describes the account component entirely: its name, description, version, and storage layout.
 
-The storage layout must specify a contiguous list of slot values that starts at index `0`, and can optionally specify initial values for each of the slots. Alternatively, placeholders can be utilized to identify values that should be provided at the moment of instantiation.
+The storage layout is described as a set of named storage slots. Each slot name must be a valid `StorageSlotName`, and its slot ID is
+derived deterministically from the name.
+
+A slot can either define a concrete value (optionally containing typed fields filled at instantiation),
+or declare a whole-slot type whose value is supplied at instantiation time.
```

Comment:
I think this sentence is a bit confusing. I'd rephrase it in terms of types and default values - i.e., each slot has a type and an optional default value. if there is no default value, the value for the slot needs to be provided at instantiation time.

Reply:
Rephrased as suggested